examples/Example311_HeatEquation_BoundaryDiffusion.jl

```diff
@@ -0,0 +1,131 @@
+#=
+
+# 311: Heat Equation with boundary diffusion 
+([source code](SOURCE_URL))
+
+
+=#
+
+module Example311_HeatEquation_BoundaryDiffusion
+using Printf
+using VoronoiFVM
+using ExtendableGrids
+
+"""
+  We solve the following system
+
+      ∂_tu - εΔu = 0            in [0,T] × Ω
+           ε∇u⋅ν = k(u-v)       on [0,T] × Γ_1
+           ε∇u⋅ν = 0            on [0,T] × (∂Ω ∖ Γ_1)
+  ∂_tv -ε_ΓΔ_Γ v = f(x) +k(u-v) on [0,T] × Γ_1
+          u(0)   = 0.5          in   {0} × Ω
+          v(0)   = 0.5          on   {0} × Γ_1  
+"""
+
+function main(n=1)
+
+    breg = 5 # boundary region number for surface diffusion
+
+    println("Surface diffusion example")
+
+    hmin   = 0.01 * 2.0^(-n+1)
+    hmax   = 0.1  * 2.0^(-n+1)
+    XLeft  = geomspace(0.0, 0.5, hmax, hmin)
+    XRight = geomspace(0.5, 1.0, hmin, hmax)
+    X      = glue(XLeft, XRight)
+    
+    Z      = geomspace(0.0, 1.0, hmin, 2*hmax)
+    
+    grid   = VoronoiFVM.Grid(X, X, Z)
+
+
+    # parameters
+    eps      = 1.0e0  # bulk heat conduction coefficient
+    eps_surf = 1.0e-2 # surface diffusion coefficient
+    k        = 1.0    # transmission coefficient
+    physics  = VoronoiFVM.Physics(
+        num_species = 2,
+        flux        = function(f,u0,edge)
+                            u    = unknowns(edge,u0)
+                            f[1] = eps*(u[1,1]-u[1,2])
+                      end,
+        
+        bflux       = function(f, u0, edge)
+                        u        = unknowns(edge,u0)
+                        f[2]     = eps_surf*(u[2,1]-u[2,2])
+                      end,
+        breaction   = function(f,u,node)
+                        if node.region == breg
+                          f[1] = k*(u[1]-u[2])
+                          f[2] = k*(u[2]-u[1])
+                        else
+                          f[1] = 0        
```

Comment:
Use f[1]=0.0  for type stability here. May well be the reason of the allocations.

Reply:
IT appears that the allocations happen in `eval_and_assemble` in the loop over the boundary edges when filling the edge element:
```
  edge.index   = bfaceedges[ibedge, ibface]
  edge.node[1] = bedgenodes[1, edge.index]
  edge.node[2] = bedgenodes[2, edge.index]
```